src/extractors/youtube-new/utils/errors.ts

```diff
@@ -0,0 +1,69 @@
+import { utilityErrors } from "@utils";
+
+/**
+ * @class General error class for the YouTube extractor. It is recommended to extend this class for more detailed error names.
+ */
+export class YoutubeError extends utilityErrors.VueTubeExtractorError { }
+
+/**
+ * @abstract Errors for http requests.
+ *
+ * @param {string} id - the id of the video
+ */
+abstract class PageError extends YoutubeError {
+  id: string;
+  constructor(id: string, message: string, details?: unknown) {
+    super(message, details);
+    // Object.setPrototypeOf(this, PageError.prototype);
+    this.id = id;
+  }
+}
+
+abstract class VideoError extends PageError { }
```

Comment:
We don't need these error objects anymore. Just use the ones under `src/utils/errors.ts` with an error message instead

Reply:
updated in e84daeb